platform/src/main/java/org/stellar/anchor/platform/action/RequestOnchainFundsHandler.java

```diff
@@ -0,0 +1,186 @@
+import org.stellar.anchor.horizon.Horizon;
+import org.stellar.anchor.platform.data.JdbcSep24Transaction;
+import org.stellar.anchor.platform.data.JdbcSepTransaction;
+import org.stellar.anchor.platform.service.Sep24DepositInfoNoneGenerator;
+import org.stellar.anchor.sep24.Sep24DepositInfoGenerator;
+import org.stellar.anchor.sep24.Sep24TransactionStore;
+import org.stellar.anchor.sep31.Sep31TransactionStore;
+import org.stellar.sdk.Memo;
+
+public class RequestOnchainFundsHandler extends ActionHandler<RequestOnchainFundsRequest> {
+
+  private final CustodyService custodyService;
+  private final CustodyConfig custodyConfig;
+  private final Sep24DepositInfoGenerator sep24DepositInfoGenerator;
+
+  public RequestOnchainFundsHandler(
+      Sep24TransactionStore txn24Store,
+      Sep31TransactionStore txn31Store,
+      Validator validator,
+      Horizon horizon,
+      AssetService assetService,
+      CustodyService custodyService,
+      CustodyConfig custodyConfig,
+      Sep24DepositInfoGenerator sep24DepositInfoGenerator) {
+    super(
+        txn24Store, txn31Store, validator, horizon, assetService, RequestOnchainFundsRequest.class);
+    this.custodyService = custodyService;
+    this.custodyConfig = custodyConfig;
+    this.sep24DepositInfoGenerator = sep24DepositInfoGenerator;
+  }
+
+  @Override
+  protected void validate(JdbcSepTransaction txn, RequestOnchainFundsRequest request)
+      throws InvalidParamsException, InvalidRequestException {
+    super.validate(txn, request);
+
+    if (!((request.getAmountIn() == null
+            && request.getAmountOut() == null
+            && request.getAmountFee() == null
+            && request.getAmountExpected() == null)
+        || (request.getAmountIn() != null
+            && request.getAmountOut() != null
+            && request.getAmountFee() != null))) {
+      throw new InvalidParamsException(
+          "All or none of the amount_in, amount_out, and amount_fee should be set");
+    }
+
+    validateAsset("amount_in", request.getAmountIn());
+    validateAsset("amount_out", request.getAmountOut());
+    validateAsset("amount_fee", request.getAmountFee(), true);
+    if (request.getAmountExpected() != null) {
+      validateAsset(
+          "amount_expected",
+          AmountRequest.builder()
+              .amount(request.getAmountExpected())
+              .asset(request.getAmountIn().getAsset())
+              .build());
+    }
+
+    if (request.getAmountIn() == null && txn.getAmountIn() == null) {
+      throw new InvalidParamsException("amount_in is required");
+    }
+    if (request.getAmountOut() == null && txn.getAmountOut() == null) {
+      throw new InvalidParamsException("amount_out is required");
+    }
+    if (request.getAmountFee() == null && txn.getAmountFee() == null) {
+      throw new InvalidParamsException("amount_fee is required");
+    }
+
+    if (sep24DepositInfoGenerator instanceof Sep24DepositInfoNoneGenerator) {
```

Comment:
I think you should throw if memo/destination account is provided and info generator is not "None" (cause this fields will be ignored and business should be notified about it)